src/core/proxy.ts

```diff
@@ -239,8 +239,14 @@ export function markChangedProxy(state: ImmerState) {
 		) {
 			const copy = (state.copy_ = shallowCopy(state.base_))
 			each(state.drafts_!, (key, value) => {
-				// @ts-ignore
-				copy[key] = value
+				const descProto = Object.getOwnPropertyDescriptor(
+					Object.getPrototypeOf(copy),
+					key
+				)
+				if (!(descProto && descProto.get && !descProto.set)) {
+					// @ts-ignore
+					copy[key] = value
```

Comment:
Q: If it is a getter, do we actually need to do anything at all or could we ignore the entire field?🤔 Since we should already pick up the change in the backing properties?

Reply:
Yes, that is the idea, just skip the field if it has a getter without a setter.

Another approach is to create an empty setter for these cases when creating the shallowCopy:

```js
src\utils\common.ts

export function shallowCopy(base: any, invokeGetters = false) {
	if (Array.isArray(base)) return base.slice()
+	const proto = Object.getPrototypeOf(base)
+	if (proto) {
+		Object.entries(Object.getOwnPropertyDescriptors(proto)).forEach(
+			([key, desc]) => {
+				if (desc.get && !desc.set) {
+					Object.defineProperty(proto, key, {
+						...desc,
+						set: x => {}
+					})
+				}
+			}
+		)
+	}
	const clone = Object.create(Object.getPrototypeOf(base))
```
I have tested and it also works well.

Reply:
I misread something in the original, I think that one is fine as is :)